src/controller/artifact/annotation/parser.go

```diff
@@ -0,0 +1,76 @@
+package annotation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/goharbor/harbor/src/lib/log"
+
+	"github.com/goharbor/harbor/src/pkg/artifact"
+)
+
+const (
+	// XIcon is icon content type image/x-icon
+	XIcon = "image/x-icon"
+	// GIF is icon content type image/gif
+	GIF = "image/gif"
+	// PNG is icon content type image/png
+	PNG = "image/png"
+	// JPEG is icon content type image/jpeg
+	JPEG = "image/jpeg"
+
+	// AnnotationPrefix is the prefix of annotation
+	AnnotationPrefix = "io.goharbor.artifact"
+
+	// SkipList is the key word of skip-list annotation
+	SkipList = "skip-list"
+	// Icon is the key word of icon annotation
+	Icon = "icon"
+)
+
+var (
+	// Registry for registered annotation parsers
+	Registry = map[string]Parser{}
+
+	// SortedAnnotationVersionList define the order of AnnotationParser from low to high version.
+	// Low version annotation parser will parser annotation first.
+	SortedAnnotationVersionList = make([]string, 0)
+)
+
+// NewParser creates a new annotation parser
+func NewParser() Parser {
+	return &parser{}
+}
+
+// Parser parses annotations in artifact manifest
+type Parser interface {
+	// Parse parses annotations in artifact manifest, abstracts data from artifact config layer into the artifact model
+	Parse(ctx context.Context, artifact *artifact.Artifact, manifest []byte) (err error)
+}
+
+type parser struct{}
+
+func (p *parser) Parse(ctx context.Context, artifact *artifact.Artifact, manifest []byte) (err error) {
+	for _, annotationVersion := range SortedAnnotationVersionList {
+		err = GetAnnotationParser(annotationVersion).Parse(ctx, artifact, manifest)
+	}
+	log.Error(err)
+	return
+}
+
+// RegisterAnnotationParser register annotation parser
+func RegisterAnnotationParser(parser Parser, version string) error {
+	_, exist := Registry[version]
+	if exist {
+		return fmt.Errorf("the annotation parser to parser artifact annotation version %s already exists", version)
+	}
+	Registry[version] = parser
+	SortedAnnotationVersionList = append(SortedAnnotationVersionList, version)
+	log.Infof("the annotation parser to parser artifact annotation version %s registered", version)
+	return nil
+}
+
+// GetAnnotationParser register annotation parser
+func GetAnnotationParser(version string) Parser {
```

Comment:
as we already have GetAnnotationParser to return a parser, should we private the Registry varaible?